### PR TITLE
Node grpc client

### PR DESCRIPTION
**Pull request by sushantmimani**
-  Add node gRPC client
- Add docker-compose file for node client
- Rename docker-compose file for python client

### CODE REVIEW

File: server/samples/clients/node/appencryption_client.js [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    #decryptResolve;
+
+    constructor(callback) {
+        let appEncryptionDef = protoLoader.loadSync(__dirname + protoPath, {
+            keepCase: true,
+            defaults: true,
+            oneofs: true,
+        });
+
+        let appEncryptionProto = grpc.loadPackageDefinition(appEncryptionDef);
```
Comment (aka-bo) [inline server/samples/clients/node/appencryption_client.js:48]: Does every instance need to load the proto def? If not, consider passing as an argument instead.

File: server/samples/clients/node/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+<snip>
+[grpc] Success: "/usr/local/lib/node_modules/grpc/src/node/extension_binary/node-v79-darwin-x64-unknown/grpc_node.node" is installed via remote
++ grpc-tools@1.8.1
++ grpc@1.24.2
+
+```
+
+Generate the client
+```console
+[user@machine python]$  grpc_tools_node_protoc --js_out=import_style=commonjs,binary:../node/ \
```
Comment (aka-bo) [inline server/samples/clients/node/README.md:32]: replace "python" with "node"

File: server/samples/clients/node/appencryption_client.js [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+/**
+ * Executes run_client until the process is interrupted.
+ * @param client
+ * @returns {Promise<void>}
+ */
+
+async function run_continuously(client) {
+    while (true) {
+        process.on('SIGINT', () => {
```
Comment (aka-bo) [inline server/samples/clients/node/appencryption_client.js:192]: Should handle SIGTERM here as well as it's the signal issued by docker on container stop

File: server/samples/clients/node/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+info: encrypting payload BN3bOy1zCKQs
+info: received DRR
+info: decrypting DRR
+info: received decrypted data: BN3bOy1zCKQs
+info: test completed successfully
+
+```
+
+## Statically generating the client
+Install the client tools
```
Comment (aka-bo) [inline server/samples/clients/node/README.md:19]: It's probably worth noting here that the sample client app doesn't support static cogegen. could also add a link to the static [routeguide example in the grpc repo](https://github.com/grpc/grpc/tree/v1.28.1/examples/node/static_codegen/route_guide)

File: server/samples/clients/node/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+```console
+[user@machine node]$ node appencryption_client.js
+info: starting test
+info: encrypting payload BN3bOy1zCKQs
+info: received DRR
+info: decrypting DRR
+info: received decrypted data: BN3bOy1zCKQs
+info: test completed successfully
+
```
Comment (aka-bo) [inline server/samples/clients/node/README.md:15]: extra line can be deleted

File: server/samples/clients/node/appencryption_client.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+    if (continuous) {
+        await run_continuously(client);
+    } else {
+        await run_once(client);
+    }
+}
+
+function loadProtoDef(protoPath) {
+    let appEncryptionDef = protoLoader.loadSync(__dirname + protoPath, {
```
Comment (aka-bo) [inline server/samples/clients/node/appencryption_client.js:219]: This is a bug. `__dirname + protoPath` doesn't work with absolute paths.

For example, given the repo path `/asherah`
Running `node appencryption_client.js --proto-path /asherah/server/protos/appencryption.proto` will result in an error:
```
/asherah/server/samples/clients/node/node_modules/@grpc/proto-loader/node_modules/prot
obufjs/src/root.js:94
            throw err;
            ^

Error: ENOENT: no such file or directory, open '/asherah/server/samples/clients/node/asherah/server/protos/appencryption.proto'
```

Possible fix would be to pass `protoPath` as the filename here and update L261 to use `__dirname`

File: server/samples/clients/node/appencryption_client.js [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+                type: 'num',
+                demand: false,
+                default: 1,
+            },
+            'p': {
+                alias: 'proto-path',
+                describe: 'The path to the proto file for the service',
+                type: 'string',
+                demand: false,
+                default: '../../../../protos/appencryption.proto',
```
Comment (aka-bo) [inline server/samples/clients/node/appencryption_client.js:261]: Additionally, `path.resolve()` can be used here to clean up the path is presented in the help output.
```suggestion
                default: path.resolve(__dirname + '../../../../protos/appencryption.proto'),
```

File: server/samples/clients/node/appencryption_client.js [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+    }
+
+    /**
+     * Create a new session for communicating with the gRPC server
+     * @param partition
+     * @returns {Promise<>}
+     */
+    getSession(partition) {
+        let self = this;
+        return new Promise((resolve, err) => {
```
Comment (nikoo28) [inline server/samples/clients/node/appencryption_client.js:88]: the `err` here is currently being ignored. same in L:102 and L:115

File: server/samples/clients/node/appencryption_client.js (hunk 3 above)
Comment (aka-bo) [reply server/samples/clients/node/appencryption_client.js]: The signal listeners to be moved outside of the loop. While it's currently functional you really only need one listener per signal for the process.

**Comment by nikoo28** [comment]
All my concerns have been addressed.

**Pull request merged**